File: Bubble Sort + Selection Sort + Insertion Sort/SelectionSort.js

let arr = [7 ,8, 3, 1, 2]; 

for(let i=0; i<arr.length-1; i++) { 

    let smallest = i; 
    for(j=i+1; j<arr.length; j++) { 
        if(arr[j] < arr[smallest]) {
            smallest = j;
        }
    }
    // swap 
    let temp = arr[i];
    arr[i] = arr[smallest];
    arr[smallest] = temp
}

console.log("Sorted Array : ", arr)

File: Bubble Sort + Selection Sort + Insertion Sort/py/SelectionSort.py

arr = [7, 8, 3, 1, 2]
n = len(arr) 

# selection sort
for i in range(n):
    smallest = i;  # assume current index has the minimum
    
    # Find the index of the smallest element in the remaining array
    for j in range(i+1, n):
        if(arr[j] < arr[smallest]):
            smallest = j
            
    # swap the found minimum with the element at index i
    arr[i], arr[smallest] = arr[smallest], arr[i]
    
print("Sorted Array : ", arr)